include/pybind11/stl_bind.h

```diff
@@ -592,9 +600,26 @@ class_<Map, holder_type> bind_map(handle scope, const std::string &name, Args&&.
 
     cl.def(init<>());
 
+    cl.def(init([](const Map& d) {
+        Map m;
```

Comment:
Both map (https://en.cppreference.com/w/cpp/container/map/map) and unordered_map (https://en.cppreference.com/w/cpp/container/unordered_map/unordered_map) that should be used instead of re-creating a custom copy constructor here.